src/main/java/org/verdictdb/coordinator/VerdictSingleResultFromListData.java

```diff
@@ -0,0 +1,158 @@
+package org.verdictdb.coordinator;
+
+import com.google.common.base.Optional;
+import com.rits.cloning.Cloner;
+import org.verdictdb.connection.DataTypeConverter;
+import org.verdictdb.connection.DbmsQueryResult;
+import org.verdictdb.connection.DbmsQueryResultMetaData;
+
+import java.io.*;
+import java.sql.Types;
+import java.util.List;
+
+public class VerdictSingleResultFromListData extends VerdictSingleResult {
+
+  private Optional<List<Object>> result;
+
+  // used to support wasnull()
+  private Object lastValueRead;
+
+  int cursor = -1;
+
+  public VerdictSingleResultFromListData(List<Object> result) {
+    super();
+    if (result == null) {
+      this.result = Optional.absent();
+    } else {
+      List<Object> copied = copyResult(result);
+      this.result = Optional.of(copied);
+    }
+  }
+
+  public VerdictSingleResultFromListData(List<Object> result, boolean asIs) {
+    // If result contains objects that cannot be serialized (e.g., BLOB, CLOB in H2),
+    // it is just copied as-is (i.e., shallow copy) as opposed to deep copy.
+    super();
+    if (result == null) {
+      this.result = Optional.absent();
+    } else {
+      if (asIs) {
+        this.result = Optional.of(result);
+      } else {
+        List<Object> copied = copyResult(result);
+        this.result = Optional.of(copied);
+      }
+    }
+  }
+
+  public static VerdictSingleResultFromListData empty() {
+    return new VerdictSingleResultFromListData(null);
+  }
+
+  public boolean isEmpty() {
+    return !result.isPresent();
+  }
+
+
+  private List<Object> copyResult(List<Object> result) {
+    List<Object> copied = new Cloner().deepClone(result);
+    return copied;
+  }
+
+  public DbmsQueryResultMetaData getMetaData() {
+    return null;
+  }
+
+  @Override
+  public int getColumnCount() {
+    if (result.isPresent() == false) {
+      return 0;
+    } else {
+      Object o = result.get().get(0);
+      if (o instanceof List) {
+        return ((List) o).size();
+      }
+      else return 1;
+    }
+  }
+
+  @Override
+  public String getColumnName(int index) {
```

Comment:
Please create a field that contains column names. The field should be initialized in the constructor using the given parameter.

For example, "show table" will display "table name" as a column name. "describe table" will display "column name" in the first column, and "column type" in the second column. These header names should be set by the caller (i.e., in ExecutionContext).

Reply:
I think describe table will not only return column name and column type. Shall we also include other columns?

Reply:
What does "describe table" return in addition to column name and column types? Something like primary key information?

Reply:
Yes. At least in MySQL, it will return other columns NULL, KEY, DEFAULT and EXTRA

Reply:
Our DbmsConnection does not provide that extra information though. For now, let's just return column names and types. We may need to provide more information in the future, which will require updating our DbmsConnection interface too, however.